docs/Developers/Design/deploy_procedure_design.md

```diff
@@ -0,0 +1,42 @@
+This document describes the actions that have to be executed successfully for
+monstorak's CRDs to be considered reconciled.
+
+![ActionGraph](workflow.dot.png)
+
+# Pre-requisites
+
+## Cluster monitoring deployed
+This operator expects that cluster monitoring stack is already deployed in
+k8s and all required services like prometheus and its related artifacts
+are already available.
+
+## Storage cluster deployed
+The ceph mixins (alerting rules) are meant for storage, so the operator expects
+the storage stack to be deployed and available already. The operator
+![Rook](https://github.com/rook/rook/) takes care of deployment of ceph cluster.
+
+## Monitoring Enabled For Storage
+The namespace in which storage is deployed, monitoring needs to be enabled for
+the same already. Below commands can be used to enable the required configurations
+
+```
+- oc label namespace <storage-namespoace> openshift.io/cluster-monitoring=true
+- oc policy add-role-to-user view system:serviceaccount:openshift-monitoring:prometheus-k8s -n <storage-namespace>
+```
+
+# Deploy storage mixin
+Mixins are mechanism to write alerting rules and grafana dashboard templates in
+kubernetes. The project ![ceph-mixins](https://github.com/ceph/ceph-mixins/)
```

Comment:
Should be generic. Can use `ceph` as an example or even `gluster`

Reply:
ack